### ISSUE TITLE
[LSP] Add support for range formatting

### ISSUE THREAD

**Issue by ndavd** [NomicFoundation/hardhat-vscode#528]
The lack of range formatting capabilities make the LSP incompatible with several formatting tools such as https://github.com/stevearc/conform.nvim. It would be nice if it could be added since no formatter is complete without supporting range formatting ❤️

**Comment by kanej** [comment]
Agreed. We use prettier to power our formatting. I will need to investigate whether prettier + the prettier solidity extension provide range formatting.
